arcgis-ios-sdk-samples/Geometry/Nearest vertex/NearestVertexViewController.swift

```diff
@@ -16,131 +16,137 @@ import UIKit
+        return AGSGraphic(geometry: nil, symbol: symbol)
+    }()
+    /// The graphic for the nearest coordinate point.
+    let nearestCoordinateGraphic: AGSGraphic = {
+        let symbol = AGSSimpleMarkerSymbol(style: .diamond, color: .red, size: 10)
+        return AGSGraphic(geometry: nil, symbol: symbol)
+    }()
+    /// The graphic for the nearest vertex point.
+    let nearestVertexGraphic: AGSGraphic = {
+        let symbol = AGSSimpleMarkerSymbol(style: .circle, color: .blue, size: 15)
+        return AGSGraphic(geometry: nil, symbol: symbol)
+    }()
     
-    /// A formatter to convert units for distance.
+    /// A distance formatter to format distance measurements and units.
     let distanceFormatter: MeasurementFormatter = {
         let formatter = MeasurementFormatter()
         formatter.numberFormatter.maximumFractionDigits = 1
         formatter.numberFormatter.minimumFractionDigits = 1
         return formatter
     }()
     
-    /// The point collection that defines the polygon.
-    let createdPolygon: AGSPolygon = {
-        let polygonBuilder = AGSPolygonBuilder(spatialReference: .webMercator())
-        polygonBuilder.addPointWith(x: -5991501.677830, y: 5599295.131468)
-        polygonBuilder.addPointWith(x: -6928550.398185, y: 2087936.739807)
-        polygonBuilder.addPointWith(x: -3149463.800709, y: 1840803.011362)
-        polygonBuilder.addPointWith(x: -1563689.043184, y: 3714900.452072)
-        polygonBuilder.addPointWith(x: -3180355.516764, y: 5619889.608838)
-        return polygonBuilder.toGeometry()
-    }()
-    
-    /// The graphic for the polygon, tapped point, nearest coordinate point and nearest vertex point.
-    lazy var polygonGraphic = AGSGraphic(geometry: createdPolygon, symbol: polygonFillSymbol)
-    lazy var tappedLocationGraphic = AGSGraphic(geometry: nil, symbol: tappedLocationSymbol)
-    lazy var nearestCoordinateGraphic = AGSGraphic(geometry: nil, symbol: nearestCoordinateSymbol)
-    lazy var nearestVertexGraphic = AGSGraphic(geometry: nil, symbol: nearestVertexSymbol)
-    
-    /// The map view managed by the view controller.
-    @IBOutlet weak var mapView: AGSMapView! {
-        didSet {
-            mapView.map = makeMap()
-            mapView.graphicsOverlays.add(graphicsOverlay)
-            mapView.setViewpointCenter(
-                AGSPoint(
-                    x: -4487263.495911,
-                    y: 3699176.480377,
-                    spatialReference: .webMercator()
-                ),
-                scale: 1e8
-            )
-            mapView.touchDelegate = self
-        }
-    }
+    // MARK: Methods
     
-    /// Creates a map.
-    ///
+    /// Create a map.
     /// - Returns: A new `AGSMap` object.
     func makeMap() -> AGSMap {
-        let map = AGSMap(basemapStyle: .arcGISTopographic)
+        // California zone 5 (ftUS) state plane coordinate system.
+        let statePlaneCaliforniaZone5 = AGSSpatialReference(wkid: 2229)!
```

Comment:
Since this is being used in multiple places (on line `34` as well), could it be defined as a constant?

```swift
private extension AGSSpatialReference {
    static let statePlaneCaliforniaZone5 = Self(wkid: 2229)!
}
```

Reply:
I've applied the suggestion. I cannot use `Self(wkid)` because [it only seems to work](https://forums.swift.org/t/why-error-covariant-self-type-cannot-be-referenced-from-a-stored-property-initializer/42235/7) with final classes.

Reply:
> [it seems to only work](https://forums.swift.org/t/why-error-covariant-self-type-cannot-be-referenced-from-a-stored-property-initializer/42235/7) with final classes.

I forgot about that. 😔